src/multi_threaded.rs

```diff
@@ -236,6 +236,81 @@ where
         }
     }
 
+    /// Get the key for a static string, interning it if it does not yet exist
+    ///
+    /// This will not reallocate and copy the given string but will instead just store it
+    ///
+    /// # Panics
+    ///
+    /// Panics if the key's `try_from_usize` function fails. With the default keys, this means that
+    /// you've interned more strings than it can handle. (For [`Spur`] this means that `u32::MAX - 1`
+    /// unique strings were interned)
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use lasso::ThreadedRodeo;
+    ///
+    /// let mut rodeo = ThreadedRodeo::default();
+    ///
+    /// // Interned the string
+    /// let key = rodeo.get_or_intern_static("Strings of things with wings and dings");
+    /// assert_eq!("Strings of things with wings and dings", rodeo.resolve(&key));
+    ///
+    /// // No string was interned, as it was already contained
+    /// let key = rodeo.get_or_intern_static("Strings of things with wings and dings");
+    /// assert_eq!("Strings of things with wings and dings", rodeo.resolve(&key));
+    /// ```
+    ///
+    #[inline]
+    pub fn get_or_intern_static(&self, string: &'static str) -> K {
+        self.try_get_or_intern_static(string)
+            .expect("Failed to get or intern static string")
+    }
+
+    /// Get the key for a static string, interning it if it does not yet exist
+    ///
+    /// This will not reallocate and copy the given string but will instead just store it
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use lasso::ThreadedRodeo;
+    ///
+    /// let mut rodeo = ThreadedRodeo::default();
+    ///
+    /// // Interned the string
+    /// let key = rodeo.try_get_or_intern_static("Strings of things with wings and dings").unwrap();
+    /// assert_eq!("Strings of things with wings and dings", rodeo.resolve(&key));
+    ///
+    /// // No string was interned, as it was already contained
+    /// let key = rodeo.try_get_or_intern_static("Strings of things with wings and dings").unwrap();
+    /// assert_eq!("Strings of things with wings and dings", rodeo.resolve(&key));
+    /// ```
+    ///
+    #[inline]
+    pub fn try_get_or_intern_static(&self, string: &'static str) -> Option<K> {
+        if let Some(key) = self.map.get(string) {
+            Some(*key)
+        } else {
+            let shard = self.map.determine_map(string);
+            // Safety: The indices provided by DashMap always refer to a shard in it's shards
+            let shard = unsafe { self.map.shards().get_unchecked(shard) };
+
+            if let Some(key) = shard.read().get(string) {
+                return Some(*key.get());
+            }
+
+            // Safety: The drop impl removes all references before the arena is dropped
+            let key = K::try_from_usize(self.key.fetch_add(1, Ordering::SeqCst))?;
```

Comment:
This safety comment actually isn't needed here, it's in the other function because it involves storing a string within the arena, which isn't needed here
```suggestion
            let key = K::try_from_usize(self.key.fetch_add(1, Ordering::SeqCst))?;
```